tests/x11/libreoffice/libreoffice_mainmenu_components.pm

```diff
@@ -127,6 +127,20 @@ sub run {
     type_string "calc";                                                             #open calc
     assert_and_click 'overview-office-calc';
     assert_screen 'test-oocalc-1';
+    if (!match_has_tag('ooffice-tip-of-the-day')) {
+        # Sometimes the dialog appears after a short delay
+        wait_still_screen 3;
+        assert_screen 'test-oocalc-1';
+    }
+    if (match_has_tag('ooffice-tip-of-the-day')) {
+        # Unselect "_S_how tips on startup", select "_O_k"
+        send_key "alt-s";
+        while (match_has_tag('ooffice-tip-of-the-day')) {
+            # make sure the dialog has really closed before quiting with "ctrl-q"
+            wait_screen_change { send_key "alt-o" };
+            assert_screen 'test-oocalc-1';
+        }
```

Comment:
Looks like this loop can be replaced by `send_key_until_needlematch 'test-oocalc-1', 'alt-o';`

Reply:
but its 'send_key_while_needlematch` ...

Reply:
yes, this is why suggested to look for `test-oocalc-1` as the final picture assuming that the needle `test-oocalc-1` would only match if the dialog is not shown. If it is not the case we could create proper new needles and delete older ambiguous one.

Reply:
ah, sorry. Read your comment in before that the needle is ambiguous. Still, what do you think about making that unambiguous and deleting all needles of tag `test-oocalc-1` with the dialog shown?

Reply:
Depending on how many screen check iterations we would usually encounter it might make sense to have a short "sleep" period, e.g. `sleep .1` after each step, WDYT?